src/Psl/Result/ResultInterface.php

```diff
@@ -95,6 +95,17 @@ public function always(Closure $always): ResultInterface;
      */
     public function getResult();
 
+    /**
+     * Unwrap the Result if it is succeeded or return $default value.
+     *
+     * @param D $default
+     *
+     * @return T|D
+     *
+     * @template D
+     */
```

Comment:
template comes first

```suggestion
     * Unwrap the Result if it is succeeded or return $default value.
     *
     * @template D 
     *
     * @param D $default
     *
     * @return T|D
     */
```